### ISSUE TITLE
Add code folding to the JavaScript editor

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3479]
It would be nice if you could collapse and expand sections of the code (like functions / if statements / loops).

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3585
Reporter: narupley
Created: 2015-01-22T12:32:25.000-0800

**Comment by rbeckman-nextgen** [comment]
Done with the new code editor changes. There are also shortcut keys available to quickly collapse/expand folds. It works for any curly brace section of code, like for loops, if statements, or even just a standalone "\{ ... \}" section, which is valid in JavaScript.

In the future we may also add code folder parsers for other syntax schemes like HL7.

Imported Comment. Original Details:
Author: narupley
Created: 2015-01-28T15:38:33.000-0800

**Comment by rbeckman-nextgen** [comment]
I copy pasted a large block of code to test this new feature.

I tested all the various cold folding options using hot keys as well as the context menu. I noticed that the block most immediately following your cursor is considered to be the 'current fold'. No problems here; everything appeared to work fine. I also tried to delete an opening brace while blocks were folded/unfolded. It seems the text editor was smart enough to dynamically adjust and unfold when it detected the braces no longer matched. However, this was not the case when deleting the closing brace as the editor still recognized this incomplete block and continued to collapse it. I checked Eclipse and it seems to be consistent with the behavior of my IDE. All things considered, deleting the last brace is a non-issue since other editors seemingly behave in the same manner.

Imported Comment. Original Details:
Author: jaysenp
Created: 2015-01-28T17:52:27.000-0800
